google/cloud/cloud_testing/check_predicate_becomes_false.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_CLOUD_TESTING_CHECK_PREDICATE_BECOMES_FALSE_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_CLOUD_TESTING_CHECK_PREDICATE_BECOMES_FALSE_H_
+
+#include "google/cloud/version.h"
+#include <gmock/gmock.h>
+#include <chrono>
+#include <thread>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace cloud_testing {
+/**
+ * Verify that an assertion becomes false after some prescribed time.
+ *
+ * Test that an assertion is initially true, but eventually becomes false after
+ * some prescribed time.  Use tolerance to avoid creating flaky tests.
+ */
+template <typename Predicate>
+void CheckPredicateBecomesFalse(Predicate&& predicate,
```

Comment:
Nit: the name should tell me when/why it becomes false.  "...BecomesFalse" --> "...BecomesFalseAfterDeadline"